buildSrc/src/main/kotlin/otel.publish-conventions.gradle.kts

```diff
@@ -14,10 +27,8 @@ publishing {
         versionParts[0] += "-${release}"
         version = versionParts.joinToString("-")
       }
-      groupId = "io.opentelemetry"
       afterEvaluate {
         // not available until evaluated.
-        artifactId = base.archivesName.get()
```

Comment:
Getting rid of this and ensuring that project name == archive name == artifact ID was key here.